routers/web/repo/pull.go

```diff
@@ -1208,44 +1207,6 @@ func CompareAndPullRequestPost(ctx *context.Context) {
 	ctx.Redirect(pullIssue.Link())
 }
 
-// TriggerTask response for a trigger task request
-func TriggerTask(ctx *context.Context) {
-	pusherID := ctx.FormInt64("pusher")
-	branch := ctx.FormString("branch")
-	secret := ctx.FormString("secret")
-	if len(branch) == 0 || len(secret) == 0 || pusherID <= 0 {
-		ctx.Error(http.StatusNotFound)
-		log.Trace("TriggerTask: branch or secret is empty, or pusher ID is not valid")
-		return
-	}
-	owner, repo := parseOwnerAndRepo(ctx)
-	if ctx.Written() {
-		return
-	}
-	got := []byte(base.EncodeMD5(owner.Salt))
-	want := []byte(secret)
-	if subtle.ConstantTimeCompare(got, want) != 1 {
-		ctx.Error(http.StatusNotFound)
-		log.Trace("TriggerTask [%s/%s]: invalid secret", owner.Name, repo.Name)
-		return
-	}
-
-	pusher, err := user_model.GetUserByID(pusherID)
-	if err != nil {
-		if user_model.IsErrUserNotExist(err) {
-			ctx.Error(http.StatusNotFound)
-		} else {
-			ctx.ServerError("GetUserByID", err)
-		}
-		return
-	}
-
-	log.Trace("TriggerTask '%s/%s' by %s", repo.Name, branch, pusher.Name)
-
-	go pull_service.AddTestPullRequestTask(pusher, repo.ID, branch, true, "", "")
```

Comment:
this func if run can repair broken pulls if PullRequestTestTask was messed due to ... a restart, misconfigured pull hooks ...

so i would call this func "repair pull" - but no idea if we like to have such func in the web interface

Reply:
> this func if run can repair broken pulls if PullRequestTestTask was messed due to ... a restart, misconfigured pull hooks ...
> 
> so i would call this func "repair pull" - but no idea if we like to have such func in the web interface

Their are some other references to this function(In `Merge`, `Update` `pushUpdates`) So it's already used where it should be.